omocha-infra/src/main/java/org/omocha/infra/repository/AuctionRepositoryImpl.java

```diff
@@ -91,6 +95,93 @@ public Page<AuctionInfo.SearchAuction> getAuctionList(
 		return PageableExecutionUtils.getPage(results, pageable, countQuery::fetchOne);
 	}
 
+	@Override
+	public Page<AuctionInfo.RetrieveMyAuctions> searchMyAuctionList(Long memberId, Auction.AuctionStatus auctionStatus,
+		Pageable pageable) {
+		JPAQuery<AuctionInfo.RetrieveMyAuctions> query = queryFactory
+			.select(new QAuctionInfo_RetrieveMyAuctions(
+				auction.auctionId,
+				auction.title,
+				auction.auctionStatus,
+				auction.nowPrice,
+				auction.endDate,
+				auction.thumbnailPath
+			))
+			.from(auction) // from 절 추가
+			.where(auction.memberId.eq(memberId)
+				.and(statusEquals(auctionStatus))); // statusEquals 메서드 확인
+
+		applySorting(pageable, auction, query);
+
+		// 페이징 적용
+		List<AuctionInfo.RetrieveMyAuctions> auctions = query
+			.offset(pageable.getOffset())
+			.limit(pageable.getPageSize())
+			.fetch();
+
+		JPAQuery<Long> countQuery = queryFactory
+			.select(auction.count())
+			.from(auction) // from 절 추가
+			.where(auction.memberId.eq(memberId) // countQuery에도 where 절 추가
+				.and(statusEquals(auctionStatus))); // statusEquals 메서드 확인
+
+		return PageableExecutionUtils.getPage(auctions, pageable, countQuery::fetchOne);
+	}
+
+	@Override
+	public Page<AuctionInfo.RetrieveMyBidAuctions> getMyBidAuctionList(Long memberId, Pageable sortPage) {
+
+		JPAQuery<Long> maxBidIds = queryFactory.select(bid.bidId.max())
+			.from(bid)
+			.where(bid.buyer.memberId.eq(memberId))
+			.groupBy(bid.auction.auctionId);
+
+		JPAQuery<AuctionInfo.RetrieveMyBidAuctions> query = queryFactory
+			.select(new QAuctionInfo_RetrieveMyBidAuctions(
+				auction.auctionId,
+				auction.title,
+				auction.auctionStatus,
+				auction.thumbnailPath,
+				Expressions.cases()
+					.when(auction.auctionStatus.eq(Auction.AuctionStatus.BIDDING))
+					.then("입찰중")
+					.when(auction.conclude.isNotNull()
+						.and(auction.conclude.buyer.isNotNull())
+						.and(auction.conclude.buyer.memberId.eq(memberId))
+						.and(auction.auctionStatus.eq(Auction.AuctionStatus.CONCLUDED)
+							.or(auction.auctionStatus.eq(Auction.AuctionStatus.COMPLETED))))
+					.then("낙찰")
+					.otherwise("패찰")
+					.as("bidStatus")
+
+			))
+			.from(auction)
+			.leftJoin(bid).on(bid.auction.eq(auction))
+			.where(bid.bidId.in(maxBidIds));
```

Comment:
```suggestion
		JPAQuery<AuctionInfo.RetrieveMyBidAuctions> query = queryFactory
			.select(new QAuctionInfo_RetrieveMyBidAuctions(
				auction.auctionId,
				auction.title,
				auction.auctionStatus,
				auction.thumbnailPath,
				Expressions.cases()
					.when(auction.auctionStatus.eq(Auction.AuctionStatus.BIDDING)).then("입찰중")
					.when(conclude.buyer.memberId.eq(memberId)
						.and(auction.auctionStatus.eq(Auction.AuctionStatus.CONCLUDED)
							.or(auction.auctionStatus.eq(Auction.AuctionStatus.COMPLETED)))).then("낙찰")
					.otherwise("패찰")
					.as("bidStatus")
			))
			.from(bid)
			.where(bid.bidId.in(maxBidIds))
			.leftJoin(auction).on(bid.auction.eq(auction))
			.leftJoin(conclude).on(auction.eq(conclude.auction));
```

전체 auction을 가지고 모든 bid와 체크를 하느라 오래걸리기 때문에
bid에서 먼저 maxBidId 찾아서 그거에 해당하는 것들만 auction으로 leftjoin 하는게 좋을 것 같아요!

또 case 문에서 when에 isNotNull 걸어놓은 것들은 불필요해보여서 모두 제거하고
auction에 conclude가 join 걸리는 걸 해결하기 위해 leftjoin을 추가적으로 명시했습니다!

해당 내용 확인해주시고 재윤님도 테스트해서 반영해주시면 좋을 것 같아요~

Reply:
테스트 해봤는데 잘 작동하네요